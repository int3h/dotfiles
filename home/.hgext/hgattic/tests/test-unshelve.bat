@echo off

rem setup hgattic in a temp test directory
setlocal
cd ..
cd > tmpfile
set /p atticdir= < tmpfile
erase tmpfile
cd tests
mkdir test
cd test
call hg init
cd .hg
echo [extensions] >tmp
echo hgattic=%atticdir%\attic.py >>tmp

rem now that attic shelves hg format patches all the time a user needs to be
rem configured for tests and datetime information must be provided for the
rem shelve command
echo [ui] >>tmp
echo username=Fred Widget ^<fred@example.com^>>>tmp
echo [defaults] >>tmp
echo attic-shelve = -d "0 0" >>tmp

move tmp hgrc
cd ..

rem replace here with test content
echo a > a.txt
call hg addrem
call hg shelve -q --git a
echo b > b.txt
call hg addrem
call hg shelve -q --git b
echo ### hg st
call hg st
echo ### hg unshelve
call hg unshelve
echo ### hg st
call hg st
call hg shelve -q
echo ### hg unshelve a
call hg unshelve a
echo ### hg st
call hg st
echo ### hg unshelve b (should fail)
call hg unshelve b
echo ### hg unshelve -f b (should pass)
call hg unshelve -f b
echo ### hg st
call hg st
echo ### current applied patch
type .hg\attic\.applied
echo ### hg shelve c (should fail)
call hg shelve c
echo ### hg shelve -f c (should pass)
call hg shelve -f --git c
echo ### hg st
call hg st
echo ### contents of c
type .hg\attic\c
echo ### files in attic
for %%f in (.hg\attic\*.*) do echo %%f
echo ### call hg unshelve --delete b
call hg unshelve --delete b
echo ### call hg unshelve --delete a (should fail)
call hg unshelve --delete a
echo ### call hg unshelve --delete -f a
call hg unshelve --delete -f a
echo ### files in attic
for %%f in (.hg\attic\*.*) do echo %%f
rem end test content

rem cleanup
cd ..
rmdir /S /Q test
endlocal
