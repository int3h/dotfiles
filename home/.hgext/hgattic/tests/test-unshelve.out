 adding a.txt
adding b.txt
### hg st
### hg unshelve
patch b unshelved
### hg st
A b.txt
### hg unshelve a
patch a unshelved
### hg st
A a.txt
### hg unshelve b (should fail)
abort: cannot apply a patch over an already active patch
### hg unshelve -f b (should pass)
patch b unshelved
### hg st
A a.txt
A b.txt
### current applied patch
b
### hg shelve c (should fail)
abort: a different patch is active
### hg shelve -f c (should pass)
patch c shelved
### hg st
### contents of c
# HG changeset patch
# User Fred Widget <fred@example.com>
# Date 0 0
diff --git a/a.txt b/a.txt
new file mode 100644
--- /dev/null
+++ b/a.txt
@@ -0,0 +1,1 @@
+a 
diff --git a/b.txt b/b.txt
new file mode 100644
--- /dev/null
+++ b/b.txt
@@ -0,0 +1,1 @@
+b 
### files in attic
.hg\attic\.applied
.hg\attic\.current
.hg\attic\a
.hg\attic\b
.hg\attic\c
### call hg unshelve --delete b
patch b unshelved
patch removed
### call hg unshelve --delete a (should fail)
abort: local changes found
### call hg unshelve --delete -f a
patch a unshelved
patch removed
### files in attic
.hg\attic\.applied
.hg\attic\.current
.hg\attic\c
